api/versions.go

```diff
@@ -216,7 +216,10 @@ func (self *version1_2) HandleRequest(requestType string, request []string, m ma
 			// Get all Docker containers.
 			containers, err = m.AllDockerContainers(query)
 			if err != nil {
-				return fmt.Errorf("failed to get all Docker containers with error: %v", err)
+				if len(containers) == 0 {
```

Comment:
i'm not sure I understand why we return an error here.  Are we trying to distinguish between when docker is entirely down vs when we miss a container because of a race?  Is there a way to do that without relying on the number of containers?  If we only had 1 container on the node, then we still get 500s when we shouldn't.

Reply:
Rollback the change here, and validate error content in `AllDockerContainers` explicitly